lib/schemas.js

```diff
@@ -2,61 +2,67 @@
 
 const joi = require('joi');
 
-const FIELD = joi.string().min(1);
-
-const ELASTICSEARCH_SCHEMA = joi.object({
-  bulk: joi.object({
-    body: joi.any().forbidden()
-  }).optional().unknown()
-}).unknown();
-
-const HANDLER_OPTIONS = joi
-  .object({
-    elasticsearch: ELASTICSEARCH_SCHEMA,
-    es: ELASTICSEARCH_SCHEMA,
-    beforeHook: joi.func(),
-    afterHook: joi.func(),
-    recordErrorHook: joi.func(),
-    errorHook: joi.func(),
-    transformRecordHook: joi.func(),
-    separator: joi.string().allow(''),
-    idField: [FIELD, joi.array().min(1).items(FIELD)],
-    idResolver: joi.func(),
-    index: joi.string().min(1),
-    indexField: [FIELD, joi.array().min(1).items(FIELD)],
-    indexPrefix: joi.string().allow(''),
-    type: joi.string().min(1),
-    typeField: [FIELD, joi.array().min(1).items(FIELD)],
-    parentField: FIELD,
-    pickFields: [FIELD, joi.array().min(1).items(FIELD)],
-    versionField: FIELD,
-    versionResolver: joi.func(),
-    retryOptions: joi.object()
-  })
-  .without('elasticsearch', 'es')
-  .oxor('idField', 'idResolver')
-  .oxor('versionField', 'versionResolver')
-  .xor('index', 'indexField')
-  .xor('type', 'typeField')
-  .without('index', 'indexPrefix')
-  .with('indexPrefix', 'indexField')
-  .label('options');
-
-const EVENT = joi.object({
-  Records: joi.array().items(joi.object({
-    eventName: joi.string().required(),
-    dynamodb: joi.object({
-      Keys: joi.object().required(),
-      NewImage: joi.object(),
-      OldImage: joi.object()
-    }).required()
-  })).required()
-});
-
-const VERSION = joi.number().min(0);
-
-module.exports = {
-  HANDLER_OPTIONS,
-  EVENT,
-  VERSION
+module.exports = function(options = {}) {
+  const elasticsearchOpts = options.elasticsearch || options.es || {};
+  const apiVersion = Number(elasticsearchOpts.apiVersion) || 5;
+
+  const FIELD = joi.string().min(1);
+
+  const ELASTICSEARCH_SCHEMA = joi.object({
```

Comment:
This schema can be extended to have an optional `apiVersion` parameter with a default set to `'5'`

Reply:
Or even better: don't set a default and treat missing as v5 (type required)